src/redux/sagas/motions/editColonyMotion.ts

```diff
@@ -0,0 +1,265 @@
+    // if (colonyAvatarImage && hasAvatarChanged) {
+    //   colonyAvatarIpfsHash = yield call(
+    //     ipfsUploadWithFallback,
+    //     getStringForColonyAvatarImage(colonyAvatarImage),
+    //   );
+    // }
+
+    /*
+     * Upload colony metadata to IPFS
+     */
+    // let colonyMetadataIpfsHash = null;
+    // colonyMetadataIpfsHash = yield call(
+    //   ipfsUploadWithFallback,
+    //   getStringForMetadataColony({
+    //     colonyDisplayName,
+    //     colonyAvatarHash: hasAvatarChanged
+    //       ? colonyAvatarIpfsHash
+    //       : colonyAvatarHash,
+    //     colonyTokens,
+    //   }),
+    // );
+
+    /**
+     * @NOTE: In order for the ColonyMetadata event (which is the only event associated with Colony Edit action) to be emitted,
+     * the second parameter must be non-empty.
+     * It will be replaced with the IPFS hash in due course.
+     */
+
+    const encodedAction = colonyClient.interface.encodeFunctionData(
+      'editColony(string)',
+      ['.'],
+    );
+
+    // create transaction
+    yield fork(createTransaction, createMotion.id, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createMotion',
+      identifier: colonyAddress,
+      params: [
+        Id.RootDomain,
+        childSkillIndex,
+        ADDRESS_ZERO,
+        encodedAction,
+        key,
+        value,
+        branchMask,
+        siblings,
+      ],
+      group: {
+        key: batchKey,
+        id: metaId,
+        index: 0,
+      },
+      ready: false,
+    });
+
+    // if (annotationMessage) {
+    //   yield fork(createTransaction, annotateEditColonyMotion.id, {
+    //     context: ClientType.ColonyClient,
+    //     methodName: 'annotateTransaction',
+    //     identifier: colonyAddress,
+    //     params: [],
+    //     group: {
+    //       key: batchKey,
+    //       id: metaId,
+    //       index: 1,
+    //     },
+    //     ready: false,
+    //   });
+    // }
+
+    yield takeFrom(createMotion.channel, ActionTypes.TRANSACTION_CREATED);
+    // if (annotationMessage) {
+    //   yield takeFrom(
+    //     annotateEditColonyMotion.channel,
+    //     ActionTypes.TRANSACTION_CREATED,
+    //   );
+    // }
+
+    yield put(transactionReady(createMotion.id));
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      createMotion.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+    yield takeFrom(createMotion.channel, ActionTypes.TRANSACTION_SUCCEEDED);
+
+    const existingTokenAddresses = getExistingTokenAddresses(colony);
+    const modifiedTokenAddresses = getPendingModifiedTokenAddresses(
+      colony.nativeToken.tokenAddress,
+      existingTokenAddresses,
+      tokenAddresses,
+    );
+
+    const haveTokensChanged = !!(
+      modifiedTokenAddresses.added.length ||
+      modifiedTokenAddresses.removed.length
+    );
+
+    /*
+     * Save the pending colony metadata in the database
+     */
+    if (colony.metadata) {
```

Comment:
Here we upload the intented changes to the metadata state ("pending" state) to the db, so it can be accessed in the block ingestor when the motion is finalized.